libs/application/ui-shell/src/hooks/useHistorySync.ts

```diff
@@ -39,30 +41,31 @@ export const useHistorySync = (
 
   // Act on history state.
   useEffect(() => {
-    if (!history) {
-      return
-    }
-    const location = `${history.location.pathname}${history.location.search}${history.location.hash}`
-    if (lastHistoryState.historyReason === 'navigate') {
-      history.push(location, lastHistoryState)
-    } else if (lastHistoryState.historyReason === 'initial') {
-      history.replace(location, lastHistoryState)
+    const url = `${location.pathname}${location.search}${location.hash}`
+    const { historyReason } = lastHistoryState
+
+    if (historyReason === 'navigate' || historyReason === 'initial') {
+      navigate(url, {
+        state: lastHistoryState,
+        replace: historyReason === 'initial',
+      })
     }
-  }, [history, lastHistoryState])
+  }, [
+    lastHistoryState,
+    location.hash,
+    location.pathname,
+    location.search,
+    navigate,
+  ])
 
-  // Listen for browser navigation.
+  // Listen for browser navigation change.
   useEffect(() => {
-    if (!history) {
-      return
+    if (
+      navigationType === 'POP' &&
+      location.state !== null &&
+      location.state.screen !== null
+    ) {
+      dispatch({ type: ActionTypes.HISTORY_POP, payload: location.state })
     }
-    return history.listen(({ state: payload }, action) => {
-      if (
-        action === 'POP' &&
-        payload?.screen != null &&
-        payload?.state != null
-      ) {
-        dispatch({ type: ActionTypes.HISTORY_POP, payload })
-      }
-    })
-  }, [history, dispatch])
+  }, [navigationType, dispatch, location.state])
```

Comment:
Yes but either way since we use location.state in the effect it should be added as a dependency. I will check and verify better on Monday when IDS on dev is working.